src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -21,6 +21,7 @@ internal class RemoteCredentialsDecorator : ICredentialsDecorator {
         private readonly AsyncReaderWriterLock _lock;
         private bool _credentialsAreValid;
         private readonly string _workspaceName;
+        private object _credentialsLock;
```

Comment:
`InvalidateCredentials` is already called inside the lock, and `GetCredential` belongs to the framework interface, so I'm not sure if it is safe to put the lock inside it.